twilio/base/client_base.py

```diff
@@ -48,9 +52,8 @@ def __init__(
         """ :type : str """
         self.user_agent_extensions = user_agent_extensions or []
         """ :type : list[str] """
-
-        if not self.username or not self.password:
-            raise TwilioException("Credentials are required to create a TwilioClient")
+        self.credential_provider = credential_provider or None
```

Comment:
Check this with Kridai, if existing customers use TwilioException - this is a breaking change

Reply:
Check if the exception being thrown as 401 is getting wrapped in TwilioException and being sent to customer

Reply:
For the below cases
- No username password provided
- Wrong username password provided
TwilioRestException is thrown now

Is this a breaking change?